Bottom: efe78ecff2e5945d7fa73be9f552d7b0e0a0df45
Top:    ceb35731b4a2b09edff6b36b831ea8cd679cf821
Author: Andrew Cooper <andrew.cooper@novatechweb.com>
Date:   2018-05-22 15:55:57 -0500

Fix wiki mysql database provision


---
diff --git a/roles/docker-mediawiki/defaults/main.yml b/roles/docker-mediawiki/defaults/main.yml
index a444eac8..8a608ebd 100644
--- a/roles/docker-mediawiki/defaults/main.yml
+++ b/roles/docker-mediawiki/defaults/main.yml
@@ -18,7 +18,7 @@ wiki_image_repo: https://github.com/novatechweb/docker-mediawiki.git
 
 # the name of the image being duilt and used for the container
 wiki_image_name: '{{ docker_registry_username }}/wiki'
-wiki_db_image_name: '{{ docker_registry_username }}/mysql'
+wiki_db_image_name: 'mysql:5'
 
 # the name of the container being started
 wiki_container_name: '{{ docker_name_prefix }}wiki'
diff --git a/roles/docker-mediawiki/tasks/main.yml b/roles/docker-mediawiki/tasks/main.yml
index 7028fa6d..81694f34 100644
--- a/roles/docker-mediawiki/tasks/main.yml
+++ b/roles/docker-mediawiki/tasks/main.yml
@@ -94,36 +94,39 @@
   docker_volume:
     name: '{{ wiki_db_dv_name }}'
 
-- name: stop prev container (mysql)
-  docker_container:
-    image: '{{ wiki_db_image_name }}'
-    name: '{{ wiki_db_container_name }}'
-    state: stopped
-
-- name: initial populate (mysql)
-  docker_db_init:
-    image_name: '{{ wiki_db_image_name }}'
-    image_tag: '{{ docker_image_tag }}'
-    container_name: '{{ wiki_db_dv_name }}_init_db'
-    data_volume_container_name: '{{ wiki_db_dv_name }}'
-    database_name: wikidb
-    database_user: '{{ wiki_db_user }}'
-    database_password: '{{ wiki_db_password | quote }}'
-    database_root_password: '{{ wiki_db_root_password | quote }}'
-
 # *****************************************************************************
-# Start the data container running
+# Start the database
 
 - name: start container (mysql)
   docker_container:
     name: '{{ wiki_db_container_name }}'
     detach: true
     restart_policy: always
-    volumes_from: '{{ wiki_db_dv_name }}'
-    image: '{{ wiki_db_image_name }}:{{ docker_image_tag }}'
+    image: '{{ wiki_db_image_name }}'
+    volumes:
+      - '{{ wiki_db_dv_name }}:/var/lib/mysql:z'
+    env:
+      MYSQL_DATABASE: 'wikidb'
+      MYSQL_ROOT_PASSWORD: '{{ wiki_db_root_password }}'
+      MYSQL_USER: '{{ wiki_db_user }}'
+      MYSQL_PASSWORD: '{{ wiki_db_password }}'
     networks:
       - name: '{{ docker_network_frontend }}'
 
+- name: wait for mysql initialization to complete
+  shell: >
+      printf 'SHOW GLOBAL STATUS\n' | docker exec -i 
+      '{{ wiki_db_container_name }}'
+      mysql --host=localhost --user={{ wiki_db_user | quote }}
+      --password={{ wiki_db_password | quote }} wikidb
+  register: container_status
+  retries: 20
+  delay: 3
+  until: "container_status.rc == 0"
+
+# *****************************************************************************
+# Start the data container running
+
 - name: start container (mediwiki)
   docker_container:
     name: '{{ wiki_container_name }}'
@@ -133,9 +136,9 @@
     ports: '{{ wiki_port_args }}'
     networks:
       - name: '{{ docker_network_frontend }}'
-    volumes-from: 
-      - '{{ openssl_dv_name }}'
-      - '{{ wiki_dv_name }}'
+    volumes: 
+      - '{{ openssl_dv_name }}:/etc/ssl/private:z'
+      - '{{ wiki_dv_name }}/var/www/html:z'
     env:
       WIKI_HOSTNAME: '{{ container_addr_map.wiki.hostname }}'
       WIKI_MAIL_USER: 'wiki'
